File: 프로그래머스/1/12932. 자연수 뒤집어 배열로 만들기/자연수 뒤집어 배열로 만들기.java
import java.util.*;

class Solution {
    public int[] solution(long n) {
        List<Integer> arr = new ArrayList<Integer>();
        while (n>0){
            arr.add(Long.valueOf(n%10).intValue());
            n/=10;
        }
        
        int[] answer = new int[arr.size()];
        for (int i = 0; i < arr.size(); i++){
            answer[i] = arr.get(i);
        }
        return answer;
    }
}

File: 프로그래머스/lv1/12932. 자연수 뒤집어 배열로 만들기/자연수 뒤집어 배열로 만들기.kt
class Solution {
    fun solution(n: Long): IntArray {
        var answer = ArrayList<Int>()
        var input:Long = n

        while(input!=0L){
            answer.add((input%10L).toInt())
            input/=10L
        }
        
        return answer.toIntArray()
    }
}